Sources/web3swift/Web3/Web3+MutatingTransaction.swift

```diff
@@ -12,25 +12,17 @@ fileprivate typealias PromiseResult = PromiseKit.Result
 public class WriteTransaction: ReadTransaction {
 
     public func assemblePromise(transactionOptions: TransactionOptions? = nil) -> Promise<EthereumTransaction> {
-        var assembledTransaction: EthereumTransaction = self.transaction
-        let queue = self.web3.requestDispatcher.queue
+        var assembledTransaction: EthereumTransaction = transaction
+        let queue = web3.requestDispatcher.queue
         let returnPromise = Promise<EthereumTransaction> { seal in
-            if self.method != "fallback" {
-                let m = self.contract.methods[self.method]
-                if m == nil {
+            if method != "fallback" {
+                guard let method = contract.methods[method]?.first else {
                     seal.reject(Web3Error.inputError(desc: "Contract's ABI does not have such method"))
                     return
                 }
-                switch m! {
-                case .function(let function):
-                    if function.constant {
-                        seal.reject(Web3Error.inputError(desc: "Trying to transact to the constant function"))
-                        return
-                    }
-                case .constructor(_):
-                    break
-                default:
-                    seal.reject(Web3Error.inputError(desc: "Contract's ABI does not have such method"))
```

Comment:
Why are we dropping iterating through method kinds? Like I see implementation of constant case, but what about function and constructor?

Reply:
Because `contract.methods` contains `ABI.Element.Function` type now instead of `ABI.Element`. Now there is no need to check what we get from `methods`.
And to make things even more reasonable - `methods` even when it was declared as `[String: ABI.Element]` and not `[String: [ABI.Element.Function]]` it always contained only `ABI.Element.Function` because of the implementation. 

See here: https://github.com/skywinder/web3swift/pull/587/files/1021b4436b64836ef638541a2d9e27d9a3e15de0#diff-312cc2258c84a5bf0b63c24357f0dffd7671216a19e389268ca0d1be173241d7L40

If `ABI.Element` is anything else but a `case .function` (which holds as a value type `ABI.Element.Function`) it is dropped/not added to the `methods`.

Reply:
<img width="675" alt="Screenshot 2022-07-09 at 16 16 15" src="https://user-images.githubusercontent.com/36865532/178107397-a72b229f-bf9f-4b56-807d-ad17e8269c4c.png">